grouper/grouper.py

```diff
@@ -61,6 +61,8 @@ async def consume_message(self, msg: ConsumerRecord, unlock: asyncio.BoundedSema
 
         inventory_id = msg_dict["host"]["id"]
         org_id = msg_dict["host"]["org_id"]
+        reporter = msg_dict["host"]["reporter"]
```

Comment:
Also any message sent from `listener/advisor_processor.py` doesn't contain the reporter field causing a crash.

Few options here:
- not send the reporter from `listener/advisor_processor.py`, it'll fallback to "" here
- send empty reporter from `listener/advisor_processor.py`, result will be same but at least schema of the message will be consistent
- try to parse reporter from advisor message in `listener/advisor_processor.py` `_parse_input_metadata` (similarly as is in inventory_processor.py) method and send it

Reply:
I would suggest parsing the reporter from advisor message. Also another possibility is to hardcode the "inventory" reporter into the messages sent from `advisor_processor`, since these messages are always triggered by inventory.